### PR TITLE
Use epoll instead of negative events for DNS

### PR DESCRIPTION
**Pull request by duke8253**

### CODE REVIEW

**Comment by shinrich** [comment]
@zizhong could you review too?

**Comment by duke8253** [comment]
@zizhong I just tested and it happens without the change as well, and my changes does not touch that part of logic, it only removes the periodic check and puts the check in `DNSConnection::trigger` where things are written to the `triggered` list. But I'll see what is causing this to happen.

File: iocore/dns/DNS.cc [hunk 1]
```diff
@@ -959,6 +957,13 @@ DNSHandler::mainEvent(int event, Event *e)
     write_dns(this);
   }
 
+  for (int i = 0; i < n_con; i++) {
```
Comment (SolidWallOfCode) [inline iocore/dns/DNS.cc]: Possibly
```
if (std::any_of(ns_down, ns_dount + n_con, [](int f) { return f != 0 })) {
  this_ethread()->schedule_at(...);
}
```

File: iocore/dns/DNSConnection.cc [hunk 2]
```diff
@@ -78,6 +78,7 @@ void
 DNSConnection::trigger()
 {
   handler->triggered.enqueue(this);
+  handler->handleEvent(0, nullptr);
```
Comment (SolidWallOfCode) [inline iocore/dns/DNSConnection.cc]: What does that do? Maybe a tiny itty bitty comment?

File: iocore/dns/DNS.cc (hunk 1 above)
Comment (duke8253) [reply iocore/dns/DNS.cc]: this looks neat

**Comment by duke8253** [comment]
[approve ci autest]

File: iocore/dns/DNSConnection.cc [hunk 3]
```diff
@@ -78,6 +78,12 @@ void
 DNSConnection::trigger()
 {
   handler->triggered.enqueue(this);
+
+  // The handler should be pionting to DNSHandler::mainEvent.
```
Comment (SolidWallOfCode) [inline iocore/dns/DNSConnection.cc]: That's obvious. The question, is, what is accomplished by calling the main handler? Why does it need to be called from here, when it didn't before?

**Comment by zwoop** [comment]
[approve ci clang-format]

**Comment by SolidWallOfCode** [comment]
We're waiting for @zizhong to confirm the update fixes the problem he observed. @duke8253 was able to replicate the problem and changed the PR to fix it.

**Comment by SolidWallOfCode** [comment]
I don't think it's a backport candidate. It's more of a cleanup. I mainly wanted to get it in for 8.0.

**Pull request merged**